fud2/tests/tests.rs

```diff
@@ -70,23 +84,77 @@ fn req_slug(driver: &Driver, req: &Request) -> String {
     desc
 }
 
-fn test_emit(driver: &Driver, req: Request) {
+fn test_emit(driver: &Driver, req: Request, tag: &str) {
     let desc = req_desc(driver, &req);
     let slug = req_slug(driver, &req);
     let ninja = emit_ninja(driver, req);
     insta::with_settings!({
         description => desc,
         omit_expression => true,
-        snapshot_suffix => slug,
+        snapshot_suffix => format!("{slug}{tag}"),
     }, {
         insta::assert_snapshot!(ninja);
     });
 }
 
+#[test]
+fn all_ops() {
+    let driver = test_driver();
+    for op in driver.ops.values() {
+        let req = fud_core::exec::Request {
+            start_file: None,
+            start_state: op.input,
+            end_file: None,
+            end_state: op.output,
+            through: vec![],
+            workdir: ".".into(),
+        };
+        test_emit(&driver, req, &format!("__op_{}", op.name));
```

Comment:
I think there may be a minor problem here with reusing `test_emit`: namely, when there is more than one op with the same pair of input/output states. I guess a simple solution here would be to specify the op we're interested in via `through`, but we could also consider a different function (similar to `test_emit` but different) that avoids doing the path-search altogether and just specifies a `Plan` object directly. The latter might be less error prone and is not very hard to do… something like this:

```rs
Plan {
  start: "input",
  steps: vec![(op, "output")],
  workdir: ".",
  stdin: false,
  stdout: false,
}
```